Misc/NEWS.d/next/Core and Builtins/2024-04-11-21-17-23.gh-issue-117431.ZxdAFN.rst

```diff
@@ -0,0 +1 @@
+Improve performance of :func:`str.startswith` and :func:`str.endswith`.
```

Comment:
BTW, did you also measure the performance of `str.removeprefix` and `str.removesuffix`?

Reply:
They also benefit! The `removeprefix` and and `removesuffix` are just a direct call to `tailmatch` and then (when needed) slicing the string. Here is a small benchmark, other cases for `removeprefix` will be similar to the corresponding cases of `startswith`.
```
x.removeprefix('a'): Mean +- std dev: [main_removeprefix] 47.3 ns +- 2.0 ns -> [pr_removeprefix] 40.2 ns +- 1.8 ns: 1.18x faster
x.removeprefix('@'): Mean +- std dev: [main_removeprefix] 28.7 ns +- 1.3 ns -> [pr_removeprefix] 26.6 ns +- 1.1 ns: 1.08x faster
x.removeprefix('abcde'): Mean +- std dev: [main_removeprefix] 48.6 ns +- 1.7 ns -> [pr_removeprefix] 44.2 ns +- 0.6 ns: 1.10x faster

Geometric mean: 1.12x faster
```
Implementation here: https://github.com/python/cpython/blob/e3ed574f6acddf76e96a9b0fcbad35f15508cba1/Objects/unicodeobject.c#L12216

Reply:
Very cool!